DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/Extensions/DocumentExtensions.cs

```diff
@@ -17,13 +18,26 @@ public static List<AutocadRootObject> GetObjects(this Document doc, IEnumerable<
         if (long.TryParse(objectIdHandle, out long parsedId))
         {
           Handle handle = new(parsedId);
-          if (doc.Database.TryGetObjectId(handle, out ObjectId myObjectId))
+          // Note: Fatal crash happens here when objects are deleted, so we need to catch it.
+          try
           {
-            if (tr.GetObject(myObjectId, OpenMode.ForRead) is DBObject dbObject)
+            if (doc.Database.TryGetObjectId(handle, out ObjectId myObjectId))
             {
-              objects.Add(new(dbObject, objectIdHandle));
+              if (tr.GetObject(myObjectId, OpenMode.ForRead) is DBObject dbObject)
+              {
+                objects.Add(new(dbObject, objectIdHandle));
+              }
             }
           }
+          catch (Autodesk.AutoCAD.Runtime.Exception e)
+          {
+            if (e.ErrorStatus == ErrorStatus.WasErased) // Note: TBD if we want to catch more things in here. For now maybe not, but it does seem like this function gets into "crashes the host app territory"
```

Comment:
Since it is in foreach loop we can maybe aggregate them and throw as global error in UI. yes, TBD